src/Makefile.am

```diff
@@ -2746,3 +2746,56 @@ tests_testEventLoop_LDADD = \
 	$(COMPAT_LIB) \
 	$(XTRA_LIBS)
 tests_testEventLoop_LDFLAGS = $(LIBADD_DL)
+
+if ENABLE_FUZZ_DECODE
+EXTRA_PROGRAMS =+ fuzzHttp1Parser fuzzRFC1035
+
```

Comment:
I am not yet convinced that the design proposed by this PR (i.e. a dedicated test program for each parsing function) is the right one because it requires building stand-alone programs to test (often rather deeply embedded/integrated) parsers. A similar design for some of the existing Squid unit tests has wasted numerous hours on maintaining the surrounding scaffolding just to make the tests compile, usually with very little payoff.

Are there any oss-fuzz integration alternatives where fuzzing does _not_ require creating dedicated unit tests and, roughly speaking, uses primary Squid code, with Squid own main() function, Squid own configuration code, etc.? If yes, what are their pros and cons compared to the proposed integration approach?

Reply:
> I am not yet convinced that the design proposed by this PR (i.e. a dedicated test program for each parsing function) is the right one because it requires building stand-alone programs to test (often rather deeply embedded/integrated) parsers. A similar design for some of the existing Squid unit tests has wasted numerous hours on maintaining the surrounding scaffolding just to make the tests compile, usually with very little payoff.

Yes, each parser requires its own dedicated fuzzer for effective testing. Combining all tests into a single program is bad. For example, if the fuzzer generates a DNS query, this query would also be parsed by the HTTP1 parser, resulting in false execution paths. This will confuse the feedback fuzzing used by OSS-Fuzz{LibFuzzer, AFL, and Honggfuzz} this will generate bad low-quality input for fuzzing.


> Are there any oss-fuzz integration alternatives where fuzzing does not require creating dedicated unit tests and, roughly speaking, uses primary Squid code, with Squid own main() function, Squid own configuration code, etc.? If yes, what are their pros and cons compared to the proposed integration approach?

Unfortunately, that can't be done.